### DISCUSSION TITLE
Allow fuzzy pickers to work in ! command

### DISCUSSION THREAD

**Discussion by yudjinn** [helix-editor/helix#10207] [General]
I find myself using some kind of fuzzy picker (fzf, charm's gum) for a lot of things. Being able to use this would be extremely helpful in helix as I'd like to paste the output into my document. I'd love the ! syntax that pipes command output into the buffer to also work with these pickers (or similar things like it). As of now it just locks up for me and gives a long error when I force close it.

**Comment by edrex** [comment]
Short of embedding a full term in helix (which I am all for btw) manipulating the shell job system using xdotool as in https://github.com/mawww/kakoune/wiki/Integrating-Other-CLI-apps is probably the only workable option for integrating external terminal interactive tools. Maybe somebody wants to get it working and document it somewhere. Might be me but I'd prefer not :smile:

**Comment by edrex** [comment]
Possibly relevant: https://github.com/helix-editor/helix/discussions/4332
(termwiz [seems to include](https://docs.rs/termwiz/latest/termwiz/terminal/index.html) a basic terminal emulator implementation?)

**Comment by edrex** [comment]
discussion of terminal interactive tool integration attempts: https://www.reddit.com/r/HelixEditor/comments/wvrjqs/ranger_file_manager_helix_amazing/
